external_services/models.py

```diff
@@ -143,52 +220,54 @@ def __str__(self):
         return out
 
     def clean(self):
-        if not self.service and not (self.menu_url and self.menu_label):
-            raise ValidationError(_("Either preconfigured service or custom URL and label needs to be provided."))
+        errors = {}
```

Comment:
That is validated by the validator on line 183

Reply:
It looks like that this clean method disallows something that is supposed to be allowed according to the help string. If I create a menu item under the Edit course page and select an existing LTI service, I can not enter any menu URL even though the help string says I could override the URL defined in the service configuration.

Reply:
Yes. LTI services never supported changing the path part and thus they had exception for menu url. Good question is if it actually should be supported and the teacher should have the power to change the path. Or if this information should be added to already super long help string.

In addition, lti services are not distinguishable in the service dropdown currently, so the teacher couldn't know if the selected service support menu_url or not. To be fair, this is probably would be good idea as more of the menu urls require service in the future.

Reply:
Right, there is the difference between a link service and an LTI service. We could allow teachers to modify the URL for both cases. Normally, the URL of the service should be used, but in special cases, the teacher could modify it while keeping the URL domain the same. 

Currently, it is indeed confusing that the dropdown in the UI does not distinguish link and LTI services since they behave differently with the menu URL parameter.

Reply:
Yep.

Though, it's important to note that in future, if teachers wish to link to mycourses for example, they need use link service and set menu_url to subpage. Then redirect page remembers the checkbox based on domain (so, student is warned about going to external site mycourses).

I don't see reason not to implement same for LTI services, but on the other hand, I don't know if any of lti services have different login points with same secret.

Reply:
I was under the impression that the LTI connections in Rubyric and ACOS exercises may vary the URL for each exercise. It would be better if they used only one launch URL for everything since that would be easier to set up and I think it is more common to use only one launch URL in the LTI world. Of course, the A+ menu items should usually not point directly to a specific exercise, but some service overview page.

Reply:
Yeah. Exercises do and they shouldn't need menu items (only ltiservice). Menu items to lti services should be for the login purposes only (e.g. backend administration to jutut and radar and student login for neuvontajono and koodisäilö).